src/libsync/propagatedownload.cpp

```diff
@@ -322,7 +330,7 @@ void GETFileJob::slotReadyRead()
             return;
         }
 
-        qint64 w = _device->write(buffer.constData(), r);
+        const qint64 w = writeToDevice(buffer.constData(), r);
```

Comment:
Same for this one

Reply:
once we start having `writeToDevice` not necessarily write bytes to a device this check no longer makes sense
I would go with a different API where write and error checking is done within `writeToDevice` it could then return a bool or something better to indicate error or success

Reply:
> once we start having `writeToDevice` not necessarily write bytes to a device this check no longer makes sense
> I would go with a different API where write and error checking is done within `writeToDevice` it could then return a bool or something better to indicate error or success

@mgallien Why would we start having this? At the moment, we are using `qint64 QIODevice::write(const char *data, qint64 maxSize)` that may already return `-1` according to documentation (https://doc.qt.io/qt-5/qiodevice.html#waitForReadyRead), or, what exactly do you mean?